kong/runloop/plugin_servers/init.lua

```diff
@@ -192,9 +192,20 @@ function get_instance_id(plugin_name, conf)
   local key = type(conf) == "table" and kong.plugin.get_id() or plugin_name
   local instance_info = running_instances[key]
 
+  local sleep_time = 0
+  local wait_count = 0
   while instance_info and not instance_info.id do
     -- some other thread is already starting an instance
-    ngx_sleep(0)
+    -- prevent busy-waiting
+    ngx_sleep(sleep_time)
+  
+    -- to prevent a potential dead loop when someone failed to release the ID
+    wait_count = wait_count + 1
+    if wait_count > 10 then
```

Comment:
the max wait count looks okay to me, but the sleep change doesn't seem necessary, as the sleep call will cause an yield. can you elaborate on it? 
for the fix itself, only the `reset_instance` should be necessary (and lgtm); let's please make sure it's in its own commit so we can do a rebase + merge (squash + merge will mix things up)

Reply:
For example, when we start the Kong, the Go plugin server needs time to start. If we waited for a while, we could expect to wait longer.
When the Go plugin server does not respond, Kong can do nothing except for waiting for more rounds, and it might quickly run out of counts.

Reply:
got your point @StarlightIbuki, just checking: is this a theoretic scenario or have you confirmed that the first (10) wait count can be exceeded and therefore the wait increase becomes useful in practice?

Reply:
@samugi each round may take less than 0.01 seconds and it's very possible for the go plugin server to take 0.1 seconds to start.

Reply:
instead of relying on loops count (which would vary in time on different hardware), does it make sense to go with a naive approach that:
- wait explictly 0.1 in every loop, check if we are good
- if have waited for more than 10 seconds, return an error